pkg/controller/storagenode/storagenode.go

```diff
@@ -234,18 +236,27 @@ func (c *Controller) syncKVDB(
 		return nil
 	}
 
-	// ensure kvdb pods are present on nodes running kvdb
-	// list kvdb nodes on the node for this cluster
-	kvdbPodList := &v1.PodList{}
-	fieldSelector := fields.SelectorFromSet(map[string]string{"nodeName": storageNode.Name})
-	err := c.client.List(context.TODO(), kvdbPodList, &client.ListOptions{
-		Namespace:     storageNode.Namespace,
-		LabelSelector: labels.SelectorFromSet(c.kvdbPodLabels(cluster)),
-		FieldSelector: fieldSelector,
-	})
+	// Do not use c.client.List API, as it would hit controller runtime cache, which does not count pod in outOfPods status,
+	// operator would keep creating kvdb pods in this case. https://portworx.atlassian.net/browse/CEE-400
+	podList, err := coreops.Instance().GetPods(cluster.Namespace, c.kvdbPodLabels(cluster))
 	if err != nil {
 		return err
 	}
+	var kvdbPods []v1.Pod
+	for _, p := range podList.Items {
+		if p.Spec.NodeName == storageNode.Name {
+			// Let's delete the pod if it's in outOfPods status, so that a new one will be created.
+			if strings.ToLower(p.Status.Reason) == "outofpods" {
+				logrus.Warningf("Found pod with outOfPods status, will delete it: %v", p)
```

Comment:
nit: `OutOfpods` status. Also could we use `%#v` to print formatted Pod, or just print the pod namespace/name?

Reply:
Namespace/name may not have enough information for debugging, as this runs into failed code path we should be ok with more info in the log. %#v requires go formatter, not sure if it's implemented, any reason in favor of it?